api/capi/src/nnstreamer-capi-single.c

```diff
@@ -418,6 +418,14 @@ ml_single_invoke (ml_single_h single,
     }
   }
 
+#ifdef ML_SINGLE_SUPPORT_TIMEOUT
+  /* Try to clear old buffer in appsink before pushing the buffer */
+  sample =
+      gst_app_sink_try_pull_sample (GST_APP_SINK (single_h->sink), GST_MSECOND * 10);
```

Comment:
How about keeping a check if the previous buffer was TIMED_OUT based on saved variable?
Current implementation adds 10ms overhead even in cases without timeout.

Reply:
Also, give a warning message that 'Next data sample might be overwritten', as this clearing of old buffer does not guarantee clearing of the old buffer.

Reply:
Add mutex-lock and handle timed-out case.